src/customer/strategies/masterpass-customer-strategy.ts

```diff
@@ -0,0 +1,119 @@
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+                if (!this._paymentMethod || !this._paymentMethod.initializationData.checkoutId) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                const cart = state.cart.getCart();
+                if (!cart) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                const { container } = masterpassOptions;
+
+                const payload = {
+                    checkoutId: this._paymentMethod.initializationData.checkoutId,
+                    allowedCardTypes: this._paymentMethod.initializationData.allowedCardTypes,
+                    amount: cart.cartAmount.toString(),
+                    currency: cart.currency.code,
+                    cartId: cart.id,
+                };
+
+                return this._masterpassScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(Masterpass => {
+                        this._signInButton = this._createSignInButton(container);
+
+                        this._signInButton.addEventListener('click', () => {
+                            Masterpass.checkout(payload);
+                        });
+                    });
+            })
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        this._paymentMethod = undefined;
+        if (this._signInButton && this._signInButton.parentNode) {
+            this._signInButton.parentNode.removeChild(this._signInButton);
+            this._signInButton = undefined;
+        }
+
+        return super.deinitialize(options);
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'In order to sign in via Masterpass, the shopper must click on "Masterpass" button.'
+        );
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        const state = this._store.getState();
+        const payment = state.payment.getPaymentId();
+
+        if (!payment) {
+            return Promise.resolve(this._store.getState());
+        }
+
+        return this._store.dispatch(
+            this._remoteCheckoutActionCreator.signOut(payment.providerId, options)
+        );
+    }
+
+    private _createSignInButton(containerId: string): HTMLElement {
+        const container = document.querySelector(`#${containerId}`);
+
+        if (!container) {
+            throw new Error('Need a container to place the button');
```

Comment:
Can you throw this error instead: `throw new InvalidArgumentError('Unable to create sign-in button without valid container ID.');`